vmm/src/device_tree.rs

```diff
@@ -165,9 +165,14 @@ mod tests {
             aver_eq!(tb, node.id, id2);
 
             // Check remove()
-            let node = device_tree.remove(&id).unwrap();
-            aver_eq!(tb, node.id, id2);
-            aver_eq!(tb, device_tree.0.len(), 0);
+            #[cfg(feature = "pci_support")]
+            {
+                let node = device_tree.remove(&id).unwrap();
+                aver_eq!(tb, node.id, id2);
+                aver_eq!(tb, device_tree.0.len(), 0);
+            }
+            // Make sure device_tree is empty because "remove" is called conditionally.
+            device_tree.0.drain();
```

Comment:
The `#[cfg(feature = "pci_support")]` from the `remove()` function should actually be removed, this would simplify everything, and this way, we don't need to conditionally check in the tests.
So I would suggest you could remove `#[cfg(feature = "pci_support")]` from the remove function and drop this PR.

Reply:
Yes. That's fine too. I think that makes more sense anyway.